app/views/recipes/_generated_shopping_list.html.erb

```diff
@@ -15,10 +17,12 @@
           <td><%= recipe_food.food.name %></td>
           <td><%= "#{ recipe_food.quantity } #{recipe_food.food.measurementUnit}"  %></td>
           <td><%= number_to_currency(recipe_food.food.price * recipe_food.quantity) %></td>
+          <% if user_signed_in? %>
           <td>
-            <%= link_to "Modify", edit_recipe_food_path %>
+            <%= link_to "Modify", edit_recipe_food_path(recipe_id: params[:id]) %>
```

Comment:
**Optional for this PR:** The Modify button is still not working on my end. (When I click on modify and fill out the form, it adds a new food instead of modifying the food that I clicked "Modify" on.) But I think we can just fix this on your next branch.